tests/integration/account/test_account.py

```diff
@@ -64,3 +64,53 @@ def get_group(display_name: str) -> Group:
 
     group = get_group(group_display_name)
     assert group
+
+
+def test_create_account_level_groups_nested_groups(
+    make_group, make_user, acc, ws, make_random, clean_account_level_groups, watchdog_purge_suffix, runtime_ctx, caplog
+):
+    suffix = f"{make_random(4).lower()}-{watchdog_purge_suffix}"
+    # Test groups:
+    # group_a (group_b, group_c)
+    # group_b (user1, user2, group_d)
+    # group_c (group_d)
+    # group_d (user3, user4)
+
+    users = list[User]()
+    for i in range(4):
+        users.append(make_user())
```

Comment:
To fix the linting errors, maybe something like this:
```suggestion
    users = [make_user(), make_user(), make_user(), make_user()]
```